docs/sriov.md

```diff
@@ -1,4 +1,66 @@
-# Enable SR-IOV on the host
+# Architecture
+
+There are multiple components involved to provide SR-IOV support in KubeVirt
+environment.
+
+* [SR-IOV device plugin](https://github.com/intel/sriov-network-device-plugin)
+  discovers available SR-IOV resources, advertises them to Kubernetes resource
+  manager, and allocates them as needed to pods and VMIs. This component
+  doesn't modify host resources, only discovers and allocates them. For VMIs,
+  allocates `vfio` device nodes to pods; also sets environment variables with
+  PCI IDs of allocated devices. These variables are later used by KubeVirt to
+  configure libvirt domain.
+* [SR-IOV CNI plugin](https://github.com/intel/sriov-cni) configures allocated
+  SR-IOV resources before a pod or VMI is started. This component modifies host
+  resources to prepare them to be used inside a pod / VMI. Executed in root
+  network namespace context. Uses netlink and other commands to configure and
+  move SR-IOV VF / PF resources into pod namespaces.
+* [Multus](https://github.com/intel/multus-cni) is a meta-plugin. It will call
+  to SR-IOV CNI plugin when VMI is attached to an SR-IOV interface. It is
+  configured through `NetworkAttachmentDefinition` CRD objects. For SR-IOV, the
+  object annotations should refer to the resource name configured inside device
+  plugin `config.json` configuration file. This reference is used by KubeVirt
+  to automatically fill in `requests` and `limits` sections of `virt-launcher`
+  pods. (The same mechanism is used for other plugins, for example, `bridge`.)
+* [OpenShift SR-IOV operator](https://github.com/openshift/sriov-network-operator)
+  configures kernel for SR-IOV, drains resources and reboots the kernel as
+  needed, deploys components listed above, and, based on values in policy CRD
+  resources managed by admin, configures the above listed components and the
+  host.
+* KubeVirt, based on values of environment variables set by SR-IOV device
+  plugin, configures libvirt domain for SR-IOV attached VMIs to use the right
+  PCI IDs. Also, fills in `requests` and `limits` sections of `virt-launcher`
+  pod spec as per attached NetworkAttachmentDefinition CRD.
+
+# Configuration
+
+> NOTE: steps to configure SR-IOV on a host are only applicable if you don't
+> use OpenShift SR-IOV operator to deploy SR-IOV components. Despite the name,
+> the operator supports installation on a plain Kubernetes cluster. The
+> operator handles most of the steps discussed below, including configuring
+> kernel parameters. The only step that cannot be handled by the operator is
+> BIOS setup, which should be handled by other means (either manually or
+> through some kind of IPMI driven automation).
+>
+> Documentation on how to use the operator
```

Comment:
Do we need to keep this as a quotation?

Reply:
I was actually looking for a way to add a top note emphasized paragraph to bring attention that the text below is not something a regular user / developer should probably use. Not sure how to do it without reusing some other syntax feature like quotations. Ideas?